random/between.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+import { defaultOptions, type RandomOptions } from "./_types.ts";
+export type { RandomOptions };
+
+/**
+ * Generates a random number between the provided minimum and maximum values.
+ *
+ * The number is in the range `[min, max)`, i.e. `min` is included but `max` is excluded.
+ *
+ * @param min - The minimum value (inclusive)
+ * @param max - The maximum value (exclusive)
+ * @param options - The options for the random number generator
+ * @returns A random number between the provided minimum and maximum values
+ *
+ * @example Usage
+ * ```ts no-assert
+ * import { randomBetween } from "@std/random";
+ *
+ * randomBetween(1, 10); // 6.688009464410508
+ * randomBetween(1, 10); // 3.6267118101712006
+ * randomBetween(1, 10); // 7.853320239013774
+ * ```
+ */
+export function randomBetween(
+  min: number,
+  max: number,
+  options?: RandomOptions,
+): number {
+  if (!isFinite(min)) {
```

Comment:
I can't honestly remember why I used `isFinite` but it might have been due to how TS types them (`Number.isFinite(number: unknown): boolean` vs `isFinite(number: number): boolean`). But I'll swap for `Number.isFinite` as the runtime safety seems more useful